### PR TITLE
[SMALLFIX] Add documentation for Go Client

### PR DESCRIPTION
**Pull request by aaudiber**

### CODE REVIEW

File: docs/en/Go-Client.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+layout: global
+title: Go Client
+nickname: Go Client
+group: Clients
```
Comment (apc999) [inline docs/en/Go-Client.md]: @gpang I don't really see this group 
http://www.alluxio.org/docs/master/en/Getting-Started.html
can you double check if there is any change missing to make clients show up there?

File: docs/en/Go-Client.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+priority: 3
+---
+
+Alluxio has a [Go Client](https://github.com/Alluxio/alluxio-go) for interacting with Alluxio through it's 
+[REST API](Clients-Rest.html). It exposes an API very similar to the native Java API. The Go client requires
```
Comment (apc999) [inline docs/en/Go-Client.md]: put hyper link for "native Java API"?

File: docs/en/Go-Client.md (hunk 1 above)
Comment (gpang) [inline docs/en/Go-Client.md]: "Clients" gets iterated over here: https://github.com/Alluxio/alluxio/blob/master/docs/_layouts/global.html#L58

File: docs/en/Go-Client.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+layout: global
+title: Go Client
+nickname: Go Client
+group: Clients
+priority: 3
+---
+
+Alluxio has a [Go Client](https://github.com/Alluxio/alluxio-go) for interacting with Alluxio through it's 
+[REST API](Clients-Rest.html). It exposes an API very similar to the native Java API. The Go client requires 
+that an Alluxio proxy service is running. See the [godoc](http://godoc.org/github.com/Alluxio/alluxio-go)
```
Comment (gpang) [inline docs/en/Go-Client.md]: NIT: If we are explicit that the proxy is required, should we also mention why (go client uses the REST api)?

File: docs/en/Go-Client.md (hunk 1 above)
Comment (aaudiber) [inline docs/en/Go-Client.md]: What Gene said. When I run `jekyll serve` I see this Go doc with the other client docs in the "Clients" section

File: docs/en/Go-Client.md (hunk 2 above)
Comment (aaudiber) [inline docs/en/Go-Client.md]: done, thanks

**Pull request merged**